components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/RequestObjectValidatorImpl.java

```diff
@@ -74,14 +76,101 @@ public boolean isSigned(RequestObject requestObject) {
     public boolean validateSignature(RequestObject requestObject, OAuth2Parameters oAuth2Parameters) throws
             RequestObjectException {
 
+        boolean isVerified;
+        Certificate certificate = null;
         SignedJWT jwt = requestObject.getSignedJWT();
-        Certificate certificate =
-                getCertificateForAlias(oAuth2Parameters.getTenantDomain(), oAuth2Parameters.getClientId());
-        boolean isVerified = isSignatureVerified(jwt, certificate);
+        try {
+             certificate =
+                    getCertificateForAlias(oAuth2Parameters.getTenantDomain(), oAuth2Parameters.getClientId());
+        } catch (RequestObjectException e) {
+            String message = "Error retrieving public certificate for service provider checking whether a jwks " +
+                    "endpoint is configured for the service provider with client_id: " + oAuth2Parameters.getClientId();
+            log.warn(message);
+            if (log.isDebugEnabled()) {
+
+                log.debug(message, e);
+            }
+        }
+        if (certificate == null) {
+            if (log.isDebugEnabled()) {
+
+                log.debug("Public certificate not configured for Service Provider with " +
+                        "client_id: " + oAuth2Parameters.getClientId() + " of tenantDomain: " + oAuth2Parameters
+                        .getTenantDomain() + ". Fetching the jwks endpoint for validating request object");
+            }
+            String jwksUri = getJWKSEndpoint(oAuth2Parameters);
+            isVerified = isSignatureVerified(jwt, jwksUri);
+        } else {
+            if (log.isDebugEnabled()) {
+
+                log.debug("Public certificate configured for Service Provider with " +
+                        "client_id: " + oAuth2Parameters.getClientId() + " of tenantDomain: " + oAuth2Parameters
+                        .getTenantDomain() + ". Using public certificate  for validating request object");
+            }
+            isVerified = isSignatureVerified(jwt, certificate);
+        }
```

Comment:
Would be better to have a debug logs to indicate the status of signature validation.